### PR TITLE
[async-await] Added generation of interceptors

### PR DESCRIPTION
**Pull request by Nekitosss**
Currently interceptors does not propagated to calls. Seems like it was forgotten to be implemented in this alpha implementation.

### CODE REVIEW

**Comment by Lukasa** [comment]
Thanks for this! Can you regenerate the various grpc examples? `make generate-echo generate-hello world generate-route-guide generate-normalization` should do it.

**Comment by Nekitosss** [comment]
> Thanks for this! Can you regenerate the various grpc examples? `make generate-echo generate-helloworld generate-route-guide generate-normalization` should do it.

Done. Async flag set only for echo generation, but I still launch whole generation. It added additional newline to the bottom of files. Not sure is it ok or not :)

**Comment by Lukasa** [comment]
The extra newlines are fine :smile:

**Comment by Lukasa** [comment]
Do you feel up to writing some tests? I'd love to see a couple of new tests, probably in `InterceptorsTests`, that validate that the interceptors continue to work correctly in async/await land.

**Comment by Nekitosss** [comment]
I decided to create separate test class for async interceptors in analogy with other async tests like `GRPCAsyncClientCallTests`. I called file `InterceptorsAsyncTests` so that its be close to default counterpart `InterceptorsTests`.
Also, writing tests directly in current `InterceptorsTests` file will cause it suffer from lots of `ifdef`s and `@available`s

Waiting for further improvements suggestion 👌

**Comment by Nekitosss** [comment]
Hello, @Lukasa! What are next steps to make further progress in this PR? Should I wait more for your review when you will have time or should I improve it somehow?

File: Tests/GRPCTests/InterceptorsAsyncTests.swift [hunk 1]
```diff
@@ -0,0 +1,173 @@
+/*
```
Comment (glbrntt) [inline Tests/GRPCTests/InterceptorsAsyncTests.swift:1]: Could you move this file to the `Tests/GRPCTests/AsyncAwaitSupport` directory?

File: Tests/GRPCTests/InterceptorsAsyncTests.swift [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import GRPC
+import HelloWorldModel
+import NIOCore
+import NIOHPACK
+import NIOPosix
+import SwiftProtobuf
+import XCTest
+
+
+#if compiler(>=5.5)
```
Comment (glbrntt) [inline Tests/GRPCTests/InterceptorsAsyncTests.swift:27]:
```suggestion
#if compiler(>=5.5)

import EchoImplementation
import EchoModel
import GRPC
import HelloWorldModel
import NIOCore
import NIOHPACK
import NIOPosix
import SwiftProtobuf
import XCTest

```

**Review by glbrntt — COMMENTED** [review]
Oops..! Thanks for catching and fixing this @Nekitosss. Would you mind running `./scripts/format.sh` to make the CI happy?

File: Tests/GRPCTests/InterceptorsAsyncTests.swift (hunk 1 above)
Comment (Nekitosss) [reply Tests/GRPCTests/InterceptorsAsyncTests.swift]: Done

**Review by glbrntt — APPROVED** [review]
Great, thanks @Nekitosss 🙏

**Pull request merged**